features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/SessionReplayFeature.kt

```diff
@@ -187,7 +188,24 @@ internal class SessionReplayFeature(
             return
         }
 
-        if (keepSession && rateBasedSampler.sample()) {
+        val recordNewSession = keepSession && rateBasedSampler.sample()
+
+        if (!initialized.get()) {
```

Comment:
I think the fix is good but in order not to duplicate logic I would extract this part of code in a:
`checkIfInitialized()` function and use it also in the `startRecording` function. The reason why we need this check also in the `startRecording` is that this method will be called later from the manual API that we will expose to the users.

Reply:
That makes sense, thanks for clarifying!

I'm thinking we might want to do other checks in the manual API as well (like checking the current session is not already tracked, maybe sampling?) and thus not expose `startRecording` directly. I'm usually not a big fan of adding code "because we'll need it later" because requirements might change by then and we might forget why we added it in the first place.
I'd be in favour of keeping `startRecording` as it is right now, but I have no strong opinion here, I can also add the check now if that's your preference.

Reply:
I would keep it like it is now, the only thing we need is to extract that code in a new function.